openmls/src/group/core_group/staged_commit.rs

```diff
@@ -278,8 +278,12 @@ impl CoreGroup {
         }
 
         let (provisional_group_epoch_secrets, provisional_message_secrets) =
-            provisional_epoch_secrets
-                .split_secrets(serialized_provisional_group_context, diff.leaf_count());
+            provisional_epoch_secrets.split_secrets(
+                serialized_provisional_group_context,
+                diff.leaf_count(),
+                // The index should be the same on TreeSync and Diff.
```

Comment:
Sounds like there should be a `debug_assert` making sure this is the case.

Reply:
On second thought, tt's probably safer to just make `own_leaf_index` on `TreeSyncDiff` crate-public and use that one.